scripts/install_environment.sh

```diff
@@ -87,6 +89,19 @@ function is_libvirtd_tcp_socket_enabled_and_running() {
     fi
 }
 
+function allow_libvirt_cross_network_traffic() {
+    # Flush LIBVIRT_FWI chain each time the network configuration is being updated
+    echo "Installing libvirt network hook to allow cross network traffic"
+    hook_dir="/etc/libvirt/hooks/network.d"
+    hook_filename="${hook_dir}/allow-cross-network-traffic.sh"
+    sudo mkdir -p "${hook_dir}"
+    sudo tee "${hook_filename}" << EOF
+#!/usr/bin/env sh
+iptables -F LIBVIRT_FWI || true
```

Comment:
Could this [simpler](https://github.com/openshift/assisted-service/blob/3cf458c0af48b42940efaf19e2a2351ae79fc60f/deploy/operator/ztp/none_platform_utils.sh#L71-L76) code be achieving the same thing as you're trying to do?

Reply:
libvirt set its own rules out of firewalld abstraction.
The `libvirt` zone in firewalld is mostly used to control guest to host traffic (and allows traffic forwarding, I guess to prevent clashes with the rules libvirt sets).

Here, flushing the chain `LIBVIRT_FWI`, managed by libvirt, allows the traffic between guest networks.

https://libvirt.org/firewall.html#firewalld-and-the-virtual-network-driver